src/OMSimulatorLib/ComponentTable.cpp

```diff
@@ -216,6 +216,56 @@ oms_status_enu_t oms::ComponentTable::getReal(const oms::ComRef& cref, double& v
   return logError("out of range (cref=" + std::string(cref) + ", time=" + std::to_string(time) + ")");
 }
 
+oms_status_enu_t oms::ComponentTable::getInteger(const oms::ComRef& cref, int& value)
+{
+  if (!resultReader)
+    logError("the table isn't initialized properly");
+
+  if (series.find(cref) == series.end())
+    series[cref] = resultReader->getSeries(cref.c_str());
+
+  for (int i=1; i<series[cref]->length; ++i)
+  {
+    if (series[cref]->time[i-1] == time)
+    {
+      value = series[cref]->value[i-1];
+      return oms_status_ok;
+    }
+    else if (series[cref]->time[i-1] <= time && series[cref]->time[i] >= time)
+    {
+      value = series[cref]->value[i-1];
```

Comment:
Interpolating an integer is not a good idea. They are only allowed to change at intervals, so returning the last computed one is always correct.

Is my understanding correct?

Reply:
yes that is correct, and this what i discussed with Lennart